src/spammer/tx_actor.rs

```diff
@@ -36,13 +37,15 @@ where
 pub struct PendingRunTx {
     tx_hash: TxHash,
     start_timestamp: usize,
+    kind: Option<String>,
 }
 
 impl PendingRunTx {
-    pub fn new(tx_hash: TxHash, start_timestamp: usize) -> Self {
+    pub fn new(tx_hash: TxHash, start_timestamp: usize, kind: Option<String>) -> Self {
```

Comment:
I'd change this to `kind: Option<&str>` -- that way the user can call it with `Some("kind")`